.github/workflows/dev.yml

```diff
@@ -0,0 +1,15 @@
+name: Unit, and lint
```

Comment:
why have all these new files instead of what we had previously? less is more IMO for such a small codebase.

Reply:
I abstracted this because the e2e tests were being run on every push and they are calling the actual api. I figured it's better to only run it on PR's to main. I can reset if that's an issue.

Reply:
gotcha, we can keep i guess. @slinlee cool w it?